### PR TITLE
Add a .gemspec file

### PR DESCRIPTION
**Pull request by andrewwu**
This adds a .gemspec file following the conventions of a newly created gem via the `bundler gem` command. The Gemfile has been moved into the .gemspec along with some other cleanup.

While this library has not been officially been published in rubygems, adding a .gemspec at least allows us to reference it in a Gemfile using

```ruby
gem 'fit', github: 'tjwallace/fit'
```
and is a step forward in addressing issue #12.

I've removed what seem to be unnecessary files. Also, the development dependencies include `yard` and `i18n` but those don't seem to be used anywhere. Are those safe to remove?

### CODE REVIEW

**Comment by doc75** [comment]
For the `yard` and `i18n` dependancy, I would let @tjwallace to explain the reason for those gems. Personnally I have no problem to remove them as I am not using any of it.

**Comment by tjwallace** [comment]
@andrewwu @doc75 to be honest I can't even remember why I added them... I'm guessing to generate documentation... I'm fine with removing them.

File: Gemfile.lock [hunk 1]
```diff
@@ -1,105 +0,0 @@
...
-  activesupport (~> 4.2.0)
-  bindata (~> 2.1.0)
-  bundler
-  i18n
-  jeweler
-  pry
-  rspec
-  rspec-its
-  simplecov
-  yard
```
Comment (tjwallace) [inline Gemfile.lock]: +1 for removing `Gemfile.lock` (http://yehudakatz.com/2010/12/16/clarifying-the-roles-of-the-gemspec-and-gemfile/)

Comment (doc75) [inline Gemfile.lock]: Good to know, thanks for the article.

**Comment by andrewwu** [comment]
@tjwallace can you please review and merge/submit to RubyGems?

Thanks!

**Comment by tjwallace** [comment]
Merged! Thanks @andrewwu. I'll now get this up to RubyGems.org

**Pull request merged**